api/src/test/java/org/openmrs/module/pacsintegration/outgoing/OrderToPacsComponentTest.java

```diff
@@ -59,40 +65,34 @@ public class OrderToPacsComponentTest extends NonTransactionalBaseModuleContextS
     @Autowired
     private OrderEventListener orderEventListener;
 
-    private RadiologyService radiologyService;
+    @Autowired
+    private OrderToPacsConverter converter;
 
     private PacsIntegrationService pacsIntegrationService;
 
-	protected final Log log = LogFactory.getLog(getClass());
+    private TestTaskRunner testTaskRunner;
 
     protected static final String XML_METADATA_DATASET = "org/openmrs/module/pacsintegration/include/pacsIntegrationTestDataset-metadata.xml";
     protected static final String XML_MAPPINGS_DATASET = "org/openmrs/module/pacsintegration/include/pacsIntegrationTestDataset-mappings.xml";
     protected static final String XML_DATASET = "org/openmrs/module/pacsintegration/include/pacsIntegrationTestDataset.xml";
 
-
-    /**
-	 * See http://listarchives.openmrs.org/Limitations-of-H2-for-unit-tests-td7560958.html . This
-	 * avoids: org.h2.jdbc.JdbcSQLException: Timeout trying to lock table "ORDERS"; SQL statement:
-	 * 
-	 * @see org.openmrs.test.BaseContextSensitiveTest#getRuntimeProperties()
-	 */
-	
-	@Before
+	@BeforeEach
 	public void setupDatabase() throws Exception {
-
+        testTaskRunner = new TestTaskRunner();
```

Comment:
By setting this testTaskRunner on the order event listener, it allows us to confirm that the listener receives a particular event with all of the expected data within it, without having to test that the logic of the OrderEventListener is correct and without needing to figure out how to make it work with different threads and contexts.